.github/renovate.json5

```diff
@@ -1,25 +1,29 @@
 {
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
-  "extends": [
-    "config:base"
+  extends: [
+    "config:recommended"
   ],
   // Don't manage versions in spock modules
   // we want to keep low versions for baseline compatibility
-  "ignorePaths": ["spock-*/**"],
-  "packageRules": [
+  ignorePaths: ["spock-*/**"],
+  packageRules: [
     // Disable major version upgrades for Groovy versions
     // as we want to keep one for each major version.
     // The same applies for mockito as we test with both 4 and 5
     {
-      matchPackagePrefixes: ["org.mockito:"],
+      matchPackagePrefixes: ["org.codehaus.groovy:"],
+      matchCurrentValue: "/^2\\./",
+      allowedVersions: "(,3.0)"
+    },
```

Comment:
Btw, Groovy 3 is the latest using an `org.codehaus.groovy` group name, so a separate rule for it is not needed.